lib/Storage.js

```diff
@@ -1,87 +1,132 @@
+XPCOMUtils.defineLazyModuleGetter(this, 'OS',
+                                  'resource://gre/modules/osfile.jsm'); /* global OS */
+XPCOMUtils.defineLazyModuleGetter(this, 'Task',
+                                  'resource://gre/modules/Task.jsm'); /* global Task */
 
-    request.onupgradeneeded = event => {
-      var db = event.target.result;
-      db.createObjectStore('storage', {keyPath: 'key'});
-    };
-  });
+const {Log} = require('./Log.js');
+
+let storePromise;
+
+function loadStorage() {
+  if (storePromise === undefined) {
+    let path = OS.Path.join(OS.Constants.Path.profileDir, 'recipe-client.json');
+    let storage = new JSONFile({path});
+    storePromise = Task.spawn(function* () {
+      yield storage.load();
+      return storage;
+    });
+  }
+  return storePromise;
+}
 
-  function makeKey(suffix) {
-    return `${prefix}.${suffix}`;
+exports.Storage = function(prefix, sandbox) {
+  if (!sandbox) {
+    throw new Error('No sandbox passed');
   }
 
   const storageInterface = {
+    /**
+     * Sets an item in the prefixed storage.
+     * @returns {Promise}
+     * @resolves With the stored value, or null.
+     * @rejects Javascript exception.
+     */
     getItem(keySuffix) {
-      return idbPromise.then(db => {
-        return new sandbox.Promise((resolve, reject) => {
-          let key = makeKey(keySuffix);
-          let transaction = db.transaction(['storage'], 'readonly');
-          let request = transaction.objectStore('storage').get(key);
-          request.onsuccess = event => {
-            if (event.target.result) {
-              resolve(event.target.result.value);
-            } else {
-              resolve(null);
-            }
-          };
-          request.onerror = err => reject(err);
+      return loadStorage()
+        .then(store => {
+          let namespace = store.data[prefix] || {};
+          const value = namespace[keySuffix] || null;
+          return sandbox.Promise.resolve(Cu.cloneInto(value, sandbox));
+        })
+        .catch(err => {
+          Log.error(err);
+          throw new Error(err.toString());
         });
-      });
     },
 
+    /**
+     * Sets an item in the prefixed storage.
+     * @returns {Promise}
+     * @resolves When the operation is completed succesfully
+     * @rejects Javascript exception.
+     */
     setItem(keySuffix, value) {
-      return idbPromise.then(db => {
-        return new sandbox.Promise((resolve, reject) => {
-          let key = makeKey(keySuffix);
-          let transaction = db.transaction(['storage'], 'readwrite');
-          let request = transaction.objectStore('storage').put({key, value});
-          request.onsuccess = () => resolve();
-          request.onerror = err => reject(err);
+      return loadStorage()
+        .then(store => {
+          if (!(prefix in store.data)) {
+            store.data[prefix] = {};
+          }
+          store.data[prefix][keySuffix] = value;
+          return store.save();
```

Comment:
I think you might want to use `saveSoon` instead of `save` here, like the docs recommend.